morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala

```diff
@@ -3448,6 +3469,57 @@ object EvaluatorMDMTests extends MorphirBaseSpec {
             )
           )
         )
+      ),
+      suite("Morphir Runtime Exception Tests")(
+        testExceptionMultiple("FailedCoercion Test")(
+          "exceptionTests",
+          "sdkAddTest",
+          List(Data.String("a"), Data.String("b"))
+        ) {
+          case TopLevelError(_, _, _) => assertTrue(true)
+          case e                      => assertNever(s"Expected FailedCoercion but $e was thrown")
+        },
+        testExceptionMultiple("UnknownTypeMismatch Test")("exceptionTests", "decimalHundred", List(Data.String("a"))) {
+          case TopLevelError(_, _, _: TypeError.UnknownTypeMismatch) => assertTrue(true)
```

Comment:
We should expect these tests to fail as we improve error messages - the wrapping structure with `TopLevelError` is messy and should change. (This isn't something that needs to be fixed in this PR, but we should be aware of this brittleness).

Reply:
Two things that are a must change in my opinion is the TopLevelError and any private exceptions like "CodeLocatedError".

And yes, I realize these tests will need to change. The implementation choice should make it relatively easy to revise and add to as the error handling changes though, which is the goal in my mind.

Reply:
Agreed on both counts - I want the functionality of `TopLevel` and `CodeLocated`, but yeah, they should not appear in the type hierarchy.